app/models/validations/sales/soft_validations.rb

```diff
@@ -37,9 +37,29 @@ def deposit_over_soft_max?
     deposit > savings * 4 / 3
   end
 
-  def hodate_3_years_or_more_saledate?
-    return unless hodate && saledate
+  def hodate_3_years_or_more_exdate?
+    return unless hodate && exdate
 
-    ((saledate.to_date - hodate.to_date).to_i / 365) >= 3
+    ((exdate.to_date - hodate.to_date).to_i / 365) >= 3
+  end
+
+  def purchase_price_out_of_soft_range?
+    return unless value && beds && la
+
+    purchase_price_range.present? && !value.between?(purchase_price_range.soft_min, purchase_price_range.soft_max)
+  end
+
+  def purchase_price_min_or_max_text
+    value < purchase_price_range.soft_min ? "minimum" : "maximum"
+  end
```

Comment:
Is there a reason why we couldn't copy the rent error format with separate `min` and `max` keys?

```yaml
    rent:
      min:
        title_text: "You told us the rent is %{brent}"
        hint_text: "The minimum rent expected for this type of property in this local authority is £%{soft_min_for_period}."
        hint_text: "The minimum rent expected for this type of property in this local authority is £%{soft_min_for_period}"
      max:
        title_text: "You told us the rent is %{brent}"
        hint_text: "The maximum rent expected for this type of property in this local authority is £%{soft_max_for_period}."
        hint_text: "The maximum rent expected for this type of property in this local authority is £%{soft_max_for_period}"
```

Reply:
that was what I initially did, this way we don't have two separate pages that contain the same question which I thought was more confusing than just dynamic text